python/grass/jupyter/testsuite/display_test.py

```diff
@@ -0,0 +1,155 @@
+        # Assert image exists
+        self.assertFileExists(grass_renderer._filename)
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_filename(self):
+        """Test that GrassRenderer creates maps with unique filenames."""
+        # Create map with unique filename
+        unique_filename = "test_filename.png"
+        grass_renderer = gj.GrassRenderer(filename=unique_filename)
+        # Add files to self for cleanup later
+        self.files.append("test_filename.png")
+        self.files.append("test_filename.grass_vector_legend")
+        # Add a vector and a raster to the map
+        grass_renderer.run("d.rast", map="elevation")
+        grass_renderer.run("d.vect", map="roadsmajor")
+        # Make sure image was created
+        self.assertFileExists("test_filename.png")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_hw(self):
+        """Test that GrassRenderer creates maps with unique height and widths."""
+        # Create map with height and width parameters
+        grass_renderer = gj.GrassRenderer(width=400, height=400)
+        # Add just a vector (for variety here)
+        grass_renderer.run("d.vect", map="roadsmajor")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_env(self):
+        """Test that we can hand an environment to GrassRenderer."""
+        # Create map with environment parameter
+        grass_renderer = gj.GrassRenderer(env=os.environ.copy())
+        # Add just a raster (again for variety)
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_text(self):
+        """Test that we can set a unique text_size in GrassRenderer."""
+        # Create map with unique text_size parameter
+        grass_renderer = gj.GrassRenderer(text_size=10)
+        # Add a vector and a raster
+        grass_renderer.run("d.vect", map="roadsmajor")
+        grass_renderer.run("d.rast", map="elevation")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_shortcut(self):
+        """Test that we can use display shortcuts with __getattr__."""
+        # Create map
+        grass_renderer = gj.GrassRenderer()
+        # Add raster to map with shortcut
+        grass_renderer.d_rast(map="elevation")
+        grass_renderer.d_vect(map="roadsmajor")
+
+    @unittest.skipIf(not can_import_folium(), "Cannot import folium")
+    def test_shortcut_error(self):
+        """Test that passing an incorrect attribute raises
+        appropriate error"""
+        # Create map
+        grass_renderer = gj.GrassRenderer()
+        # Pass bad shortcuts
+        with self.assertRaisesRegex(AttributeError, "Module must begin with 'd_'"):
+            grass_renderer.r_watersheds()
+        with self.assertRaisesRegex(AttributeError, "d.ModuleDoesNotExist"):
+            grass_renderer.d_ModuleDoesNotExist()
```

Comment:
Possibly more readable when it looks like a module name, so underscores, not camel case.